mesonbuild/build.py

```diff
@@ -210,6 +210,7 @@ class InstallDir(HoldableObject):
     subproject: str
     from_source_dir: bool = True
     install_tag: T.Optional[str] = None
+    install_dir_name: T.Optional[str] = None
```

Comment:
Maybe it would make more sense to just move this up underneath install_dir and make it mandatory in the API?